pkg/agent/inteface.go

```diff
@@ -112,8 +114,13 @@ const (
 
 	// InstallByLabel indicate to install addon based on clusters' label
 	InstallByLabel StrategyType = "LabelSelector"
+
+	// InstallByFilterFunction indicate to install addon based on a filter function
+	// If filter function returns true or it's nil, the addon will be installed to the cluster
+	InstallByFilterFunction StrategyType = "FilterFunction"
 )
 
+// Deprecated: InstallStrategy is deprecated. Please don't use it to create a InstallStrategy directly, instead, using functions like `InstallAll`, `InstallByLabel`, `InstallByFilterFunction`.
```

Comment:
@qiujian16  comment added to remind users not to use it directly.

Reply:
Sorry for the misunderstood.  I thought 'installStrategy' is safer.
And do we need to deprecate `InstallByLabel`? I thought it's convenience for users who want to use labels to filter, save they a couple of lines code.

Reply:
I even thought we may add some Types like `InstallAllExpectHostedMode` to save users code for most common cases.

Reply:
Another option is, deprecate `InstallAll` and `InstallByLabel` but add `AllPass` and `ByLabel` functions.

And the usage would be:

```go
...
WithInstallStrategy(agent.InstallWithFilterFunction("open-cluster-management-agent-addon", agent.ByLabel(seletor))).
...
```